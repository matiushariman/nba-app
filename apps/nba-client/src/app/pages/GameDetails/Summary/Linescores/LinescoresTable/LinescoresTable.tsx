import {
  selectLinescores,
  selectPeriod,
} from '../../../../../hooks/useGetBoxscore';

const LinescoresTable = () => {
  const linescores = selectLinescores();
  const period = selectPeriod();

  return (
    <table className="table-auto w-full">
      <thead className="bg-gray-300">
        <tr>
          <th className="uppercase text-xs pt-1 pb-1 pl-4" align="left">
            Team
          </th>
          {period.map((currentPeriod) => (
            <th className="uppercase text-xs" key={currentPeriod.label}>
              {currentPeriod.label}
            </th>
          ))}
          <th className="uppercase text-xs pr-4" align="right">
            Final
          </th>
        </tr>
      </thead>
      <tbody className="divide-y">
        {linescores.map((linescore) => (
          <tr key={linescore.teamTricode}>
            <td className="font-bold text-sm pt-1 pb-1 pl-4">
              {linescore.teamTricode}
            </td>
            {linescore.periods.map((period) => (
              <td
                className="text-sm"
                align="center"
                key={`${linescore.teamTricode}-${period.period}`}
              >
                {period.score}
              </td>
            ))}
            <td align="right" className="pr-4 text-sm">
              {linescore.score}
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export default LinescoresTable;
