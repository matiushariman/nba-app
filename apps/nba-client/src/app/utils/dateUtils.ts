import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
