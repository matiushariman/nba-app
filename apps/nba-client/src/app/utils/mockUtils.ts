export const shouldUseMock = process.env['NODE_ENV'] === 'development';
