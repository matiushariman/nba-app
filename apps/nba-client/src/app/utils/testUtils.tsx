import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
