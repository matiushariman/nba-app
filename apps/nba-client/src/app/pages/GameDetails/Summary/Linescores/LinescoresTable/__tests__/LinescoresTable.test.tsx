import LinescoresTable from '..';
import { render } from '../../../../../../utils/testUtils';

jest.mock('../../../../../hooks/useGetBoxscore', () => ({
  selectLinescores: jest.fn().mockReturnValueOnce([
    {
      teamTricode: 'LAL',
      score: 92,
      periods: [
        {
          period: 1,
          score: 20,
        },
        {
          period: 2,
          score: 15,
        },
        {
          period: 3,
          score: 25,
        },
        {
          period: 4,
          score: 32,
        },
      ],
    },
    {
      teamTricode: 'BOS',
      score: 131,
      periods: [
        {
          period: 1,
          score: 24,
        },
        {
          period: 2,
          score: 34,
        },
        {
          period: 3,
          score: 31,
        },
        {
          period: 4,
          score: 42,
        },
      ],
    },
  ]),
  selectPeriod: jest.fn().mockReturnValueOnce([
    {
      period: 1,
      label: 'Q1',
    },
    {
      period: 2,
      label: 'Q2',
    },
    {
      period: 3,
      label: 'Q3',
    },
    {
      period: 4,
      label: 'Q4',
    },
  ]),
}));

describe('pages/GameDetails/Summary/Linescores/LinescoresTable', () => {
  it('should render LinescoresTable that matches snapshot', () => {
    const { baseElement } = render(<LinescoresTable />);

    expect(baseElement).toMatchSnapshot();
  });
});
