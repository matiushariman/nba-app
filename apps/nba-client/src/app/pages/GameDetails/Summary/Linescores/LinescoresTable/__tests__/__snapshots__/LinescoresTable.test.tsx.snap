// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/GameDetails/Summary/Linescores/LinescoresTable should render LinescoresTable that matches snapshot 1`] = `
<body>
  <div>
    <table
      class="table-auto w-full"
    >
      <thead
        class="bg-gray-300"
      >
        <tr>
          <th
            align="left"
            class="uppercase text-xs pt-1 pb-1 pl-4"
          >
            Team
          </th>
          <th
            class="uppercase text-xs"
          >
            Q1
          </th>
          <th
            class="uppercase text-xs"
          >
            Q2
          </th>
          <th
            class="uppercase text-xs"
          >
            Q3
          </th>
          <th
            class="uppercase text-xs"
          >
            Q4
          </th>
          <th
            align="right"
            class="uppercase text-xs pr-4"
          >
            Final
          </th>
        </tr>
      </thead>
      <tbody
        class="divide-y"
      >
        <tr>
          <td
            class="font-bold text-sm pt-1 pb-1 pl-4"
          >
            LAL
          </td>
          <td
            align="center"
            class="text-sm"
          >
            20
          </td>
          <td
            align="center"
            class="text-sm"
          >
            15
          </td>
          <td
            align="center"
            class="text-sm"
          >
            25
          </td>
          <td
            align="center"
            class="text-sm"
          >
            32
          </td>
          <td
            align="right"
            class="pr-4 text-sm"
          >
            92
          </td>
        </tr>
        <tr>
          <td
            class="font-bold text-sm pt-1 pb-1 pl-4"
          >
            BOS
          </td>
          <td
            align="center"
            class="text-sm"
          >
            24
          </td>
          <td
            align="center"
            class="text-sm"
          >
            34
          </td>
          <td
            align="center"
            class="text-sm"
          >
            31
          </td>
          <td
            align="center"
            class="text-sm"
          >
            42
          </td>
          <td
            align="right"
            class="pr-4 text-sm"
          >
            131
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;
